utils/validation.py

```
@@ -0,0 +1,149 @@
+        name: The name of the parameter for error messages
+        
+    Returns:
+        float: The validated number
+        
+    Raises:
+        ValidationError: If the value is invalid
+    """
+    try:
+        num = float(value)
+    except (ValueError, TypeError):
+        raise ValidationError(f"Invalid {name}: '{value}' is not a number")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)